experiments/main.py

```diff
@@ -0,0 +1,410 @@
+                             "not fixed but rather decreases as the number of tasks observed increases (with a minimum "
+                             "of 1). If n_memories_per_class is set to a value greater than -1, the "
+                             "n_memories_per_class is used instead.")
+    parser.add_argument('--n_memories_per_class', type=int, default=-1,
+                        help="The number of samples to keep from each class, if set to -1, the total_n_memories "
+                             "argument is used instead")
+    parser.add_argument('--buffer_sampling_multiplier', type=float, default=1.0,
+                        help="A multiplier for sampling from the buffer more/less times than the size of the buffer "
+                             "(for example a multiplier of 2 samples from the buffer (with replacement) twice its size "
+                             "per epoch)")
+    parser.add_argument('--memory_strength', type=float, default=1.0,
+                        help="a weight to be multiplied by the loss from the buffer")
+    parser.add_argument('--max_mems_pool_per_class', type=int, default=1e5,
+                        help="Maximum size of the samples pool per class from which the buffer chooses the exemplars, "
+                             "use -1 for choosing from the whole class samples.")
+
+    # LUCIR Hyperparameters
+    parser.add_argument('--lucir_lambda', type=float, default=5.0,
+                        help="a weight to be multiplied by the distillation loss (only for the LUCIR method)")
+    parser.add_argument('--lucir_margin_1', type=float, default=0.5,
+                        help="The 1st margin used with the margin ranking loss for in the LUCIR method")
+    parser.add_argument('--lucir_margin_2', type=float, default=0.5,
+                        help="The 2nd margin used with the margin ranking loss for in the LUCIR method")
+
+    # Distributed arguments
+    parser.add_argument('--num_nodes', type=int, default=1,
+                        help="num of nodes to use")
+    parser.add_argument('--node_rank', default=0, type=int,
+                        help='node rank for distributed training')
+    parser.add_argument('--multiprocessing_distributed', action='store_true',
+                        help='Use multi-processing distributed training to launch '
+                             'N processes per node, which has N GPUs.')
+    parser.add_argument('--dist_url', default="env://", type=str,
+                        help='node rank for distributed training')
+
+    args = parser.parse_args()
+    args.num_workers = 6
+
+    config = prepare_config(args)
+    if "iirc" in config["dataset"]:
+        config["setup"] = IIRC_SETUP
+    else:
+        config["setup"] = CIL_SETUP
+
+    config['device'] = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    config["ngpus_per_node"] = torch.cuda.device_count()
```

Comment:
This is not a general solution. Some people might run your code on setups where they have to use specific gpus, like cuda:3 and cuda:5. To support any possible configuration, you have to allow the user to specify with args which gpus to use. Not the end of the world, but keep in mind that some people might have to modify your code to be able to run it.